core/src/mender-client.c

```diff
@@ -458,30 +458,27 @@ mender_client_exit(void) {
 
 static mender_err_t
 mender_client_work_function(void) {
-    mender_err_t ret = MENDER_OK;
-
     mender_log_info("work function: %d", mender_client_state);
 
-    if (MENDER_CLIENT_STATE_PENDING_REBOOT == mender_client_state) {
-        mender_log_info("Waiting for a reboot");
-        /* nothing to do */
-        goto END;
-    }
-
-    /* Work depending of the client state */
-    if (MENDER_CLIENT_STATE_INITIALIZATION == mender_client_state) {
-        /* Perform initialization of the client */
-        if (MENDER_DONE != (ret = mender_client_initialization_work_function())) {
-            goto END;
-        }
+    switch (mender_client_state) {
+        case MENDER_CLIENT_STATE_PENDING_REBOOT:
+            mender_log_info("Waiting for a reboot");
+            /* nothing to do */
```

Comment:
Can we add a TODO here to revisit this? We will need some sort of timer or something to _eventually_ abort the deployment if we don't get rebooted. This could be days of operation, but we have to call it a failure at some point or we risk having a brick (zombie?) device out there.

Reply:
Yup! There are more similar things, though -- for example, we might need to do a network restart after some time (RE: DHCP lease time).